### PR TITLE
PYTHON-2367 Add release automation for Python 3.9

### PR DESCRIPTION
**Pull request by ShaneHarvey**

### CODE REVIEW

**Comment by ShaneHarvey** [comment]
This is ready for review. I'll open a new ticket for adding 3.9 to macOS which will be blocked on the build team.

File: .evergreen/config.yml
Comment (prashantmital) [inline .evergreen/config.yml:821]: Is this going to get flipped back to true before merging?

Comment (ShaneHarvey) [reply .evergreen/config.yml]: Done.

**Review by prashantmital — APPROVED** [review]

**Pull request merged**
